squirrel.frontier/src/main/java/org/dice_research/squirrel/queue/scorecalculator/UriScoreCalculator.java

```diff
@@ -0,0 +1,110 @@
+package org.dice_research.squirrel.queue.scorecalculator;
+
+import org.aksw.jena_sparql_api.core.QueryExecutionFactory;
+import org.aksw.jena_sparql_api.http.QueryExecutionFactoryHttp;
+import org.apache.http.auth.AuthScope;
+import org.apache.http.auth.Credentials;
+import org.apache.http.auth.UsernamePasswordCredentials;
+import org.apache.http.client.CredentialsProvider;
+import org.apache.http.impl.client.AbstractHttpClient;
+import org.apache.http.protocol.HttpContext;
+import org.apache.jena.atlas.web.auth.HttpAuthenticator;
+import org.apache.jena.query.QueryExecution;
+import org.apache.jena.query.QuerySolution;
+import org.apache.jena.query.ResultSet;
+import org.apache.jena.sparql.core.DatasetDescription;
+import org.dice_research.squirrel.data.uri.CrawleableUri;
+import org.dice_research.squirrel.queue.scorebasedfilter.UriKeywiseFilter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.URI;
+
+/**
+ * Interface for getting duplicity score of a Uri.
+ * The duplicity score is calculated for a {@link CrawleableUri} as 1 / (number of times the Uri occurs as a subject in graphs
+ */
+public class UriScoreCalculator implements IUriScoreCalculator {
```

Comment:
Here, it is exactly the opposite to my comment in the IUriScoreCalculator interface: this class name pretends that the class calculates a score for a given URI. Although this is true, its functionality is much more specific since it calculates a score that tries to express the chance that the URI is a duplicate of an already known URI. The class name should reflect that.